codex-ui/src/vue/switch/Switch.vue

```diff
@@ -0,0 +1,104 @@
+<template>
+  <div
+    :class="[$style.switch, $style[`switch-${switchValue}-${disabled ? 'disabled' : 'default'}`]]"
+    v-on="disabled === false ? { click: changeValue } : {}"
+  >
+    <div
+      :class="[
+        $style[`switch__ellipse`],
+        $style[`switch__ellipse-${disabled ? 'disabled' : 'default'}`],
+        $style[`switch-${switchValue}`],
+      ]"
+    ></div>
+  </div>
+</template>
+
+<script lang="ts" setup>
+import { SwitchOn, Disabled } from './Switch.types.js';
+import { ref } from 'vue';
+
+const props = withDefaults(
+  defineProps<{
+    /**
+     * Value of the switch ('on' or 'off')
+     */
+    value?: SwitchOn;
+
+    /**
+     * State of interaction with the component
+     */
+    disabled?: Disabled;
+  }>(),
+  { value: true, disabled: false }
+);
+
+const switchValue = ref(props.value);
+
+/**
+ * Function for changing value of the switch on click
+ */
+function changeValue() {
+  switchValue.value = !switchValue.value;
```

Comment:
the switch component does not emit any event on value change. It will be impossible to use updated value in the host component